### PR TITLE
Add types to duration.py

### PR DESCRIPTION
**Pull request by InvincibleRMC**
Adds types to duration.py. Also adds an RCLDuration Type/Object Alias so type inference can be on the _duration_handle.

### CODE REVIEW

File: rclpy/rclpy/duration.py
Comment (fujitatomoya) [inline rclpy/rclpy/duration.py:28]:
```suggestion
DurationType = _rclpy.rcl_duration_t

```

File: rclpy/rclpy/duration.py
Comment (fujitatomoya) [inline rclpy/rclpy/duration.py:46]: How about this?

```suggestion
        self._duration_handle: DurationType = _rclpy.rcl_duration_t(total_nanoseconds)
```

File: rclpy/rclpy/duration.py
Comment (fujitatomoya) [inline rclpy/rclpy/duration.py:111]:
```suggestion
    def get_c_duration(self) -> DurationType:
```

**Comment by InvincibleRMC** [comment]
The problem with that solution is without making some custom type alias mypy, and pylance have to treat _rclpy.rcl_duration_t as type Any which weakens the type inference. For example since the _duration_handle is type Any it cannot reason about the type of the nanoseconds field.

rclpy/rclpy/duration.py:44: error: Returning Any from function declared to return "int"  [no-any-return]

```python
@property
    def nanoseconds(self) -> int:
        return self._duration_handle.nanoseconds
```

File: rclpy/rclpy/duration.py
Comment (fujitatomoya) [inline rclpy/rclpy/duration.py:111]: > The problem with that solution is without making some custom type alias mypy, and pylance have to treat _rclpy.rcl_duration_t as type Any which weakens the type inference. For example since the _duration_handle is type Any it cannot reason about the type of the nanoseconds field.
> 
> rclpy/rclpy/duration.py:44: error: Returning Any from function declared to return "int" [no-any-return]
> 
> ```python
> @property
>     def nanoseconds(self) -> int:
>         return self._duration_handle.nanoseconds
> ```

I see, thanks for the explanation. this does not break the userspace, right? (just checking) previously it returns `Any` but now `RCLDurationType`.

Comment (InvincibleRMC) [reply rclpy/rclpy/duration.py:111]: Yes previously it has the type annotation `Any`. And would now have the type annotation `RCLDurationType`. The type annotation does not have any effective on what  the actual return type is which is  `_rclpy_pybind11.Duration` I believe.  Because type annotations are ignored at run time it has no effect on the actual code behavior. I'm not entirely clear what you mean by userspace.

**Review by fujitatomoya — APPROVED** [review]

**Comment by fujitatomoya** [comment]
@InvincibleRMC thanks for working on this. the reason i do not start CI is to wait for 2nd review. once we have 2nd approval, i will start the CI. (So are other PRs)

**Comment by InvincibleRMC** [comment]
No problem. Thanks so much for the prompt reviews.

**Review by sloretz — APPROVED** [review]

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20409/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14819/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21159/)

**Pull request merged**
